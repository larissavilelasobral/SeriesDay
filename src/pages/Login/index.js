import { googleLogin, signIn } from '../../services/index.js';

export default function Login() {
  const login = document.createElement('div');
  login.innerHTML = `
    <link rel="stylesheet" href="./pages/Login/style.css" />
    
    <main class="box">
        <div class="container">

            <div class="banner">
                <img src="assets/logo.png" alt="Logo">
                <div class="title-container">
                    <h1 class="title">SeriesDay</h1>
                    <h3 class="subtitle">review de séries</h3>
                </div>
            </div>

            <section>
                <form class="form">
                <input class="input" id="email" type="email" autocomplete="on" placeholder="E-mail" required>
                <input class="input" id="password" type="password" autocomplete="on" placeholder="Senha" required>
                <p id="loginError" class="error-message font-work"></p>
                </form>

                <div class="signin">
                <button id="signin-button" class="signin-button buttons">Entrar</button>
                <p> OU </p>
                </div>

            </section>

            <button id="google-button" class="google-button buttons">
                <img class="google-icon-btn" src="../../assets/google-icon.svg" alt="Ícone do Google"/>
                Entrar com Conta Google
            </button>

            <button id="signup-button" class="signup-button buttons"> Não possui cadastro? <span>Clique aqui</span> </button>
        </div>

        <div id="modal-email" class="modal-email-container">
          <div class="modal-email">
            <h3 class="font-work"> Por favor, preencha os campos de login ou cadastre-se. </h3>
            <button id="email-confirmation" class="btn-modal btn-ok"> OK </button>
          </div>
        </div>

    </main>
    
    `;

  const email = login.querySelector('#email');
  const password = login.querySelector('#password');
  const loginError = login.querySelector('#loginError');
  const signInButton = login.querySelector('#signin-button');
  const signUpButton = login.querySelector('#signup-button');

  function startModalEmptyEmail() {
    const modalEmail = login.querySelector('#modal-email');
    if (modalEmail) {
      modalEmail.classList.add('show-modal');

      modalEmail.addEventListener('click', (e) => {
        if (e.target.id === 'modal-email' || e.target.className === 'btn-modal btn-ok') {
          modalEmail.classList.remove('show-modal');
        }
      });
    }
  }

  signInButton.addEventListener('click', (e) => {
    e.preventDefault();
    if (email.value) {
      signIn(email.value, password.value)
        .then(() => {
          window.location.hash = 'timeline'; // ir para o feed
        })
        .catch((error) => {
          const errorCode = error.code;
          if (errorCode === 'auth/invalid-email') {
            loginError.style.color = 'red"';
            loginError.innerHTML = 'Não há registro de usuário correspondente a este e-mail';
          } else if (errorCode === 'auth/wrong-password') {
            loginError.style.color = 'red';
            loginError.innerHTML = 'Senha inválida';
          }
        });
    } else {
      startModalEmptyEmail();
    }
  });

  signUpButton.addEventListener('click', (e) => {
    e.preventDefault();
    window.location.hash = 'register'; 
  });

  const googleButton = login.querySelector('#google-button');

  googleButton.addEventListener('click', (e) => {
    e.preventDefault();
    googleLogin();
  });
  return login;
}
