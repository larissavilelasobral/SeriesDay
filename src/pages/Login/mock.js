import { googleLogin, signIn } from '../../services/index.js';

export const Login = () => {
  const login = document.createElement('div');
  login.innerHTML = `
    <input class="input" id="email" type="email" autocomplete="on" placeholder="E-mail" required>
    <input class="input" id="password" type="password" autocomplete="on" placeholder="Senha" required>
    <button id="signin-button" class="signin-button buttons">Entrar</button>

    <button id="google-button" class="google-button buttons">
        <img class="google-icon-btn" src="../../assets/google-icon.svg" alt="Ícone do Google"/>
            Entrar com Conta Google
    </button>
    `;

  // VARIAVEIS
  const email = login.querySelector('#email');
  const password = login.querySelector('#password');
  const loginError = login.querySelector('#loginError');
  const signInButton = login.querySelector('#signin-button');

  // LOGIN DE USUARIOS EXISTENTES POR EMAIL E SENHA
  signInButton.addEventListener('click', (e) => {
    e.preventDefault();
    if (email.value) {
      signIn(email.value, password.value)
        .then(() => {
          window.location.hash = 'timeline'; // ir para o feed
        })
        .catch((error) => {
          const errorCode = error.code;
          if (errorCode === 'auth/invalid-email') {
            loginError.style.color = 'red"';
            loginError.innerHTML = 'Não há registro de usuário correspondente a este e-mail';
          } else if (errorCode === 'auth/wrong-password') {
            loginError.style.color = 'red';
            loginError.innerHTML = 'Senha inválida';
          }
        });
    }
  });

  //  AUTENTICAÇÃO COM CONTA GOOGLE
  const googleButton = login.querySelector('#google-button');

  googleButton.addEventListener('click', (e) => {
    e.preventDefault();
    googleLogin();
  });

  return login;
};
