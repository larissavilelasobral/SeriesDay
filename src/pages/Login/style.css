.container {
    display: inline-block;
    align-items: center;
    width: 300px;
    max-width: 700px;
    height: 30vw;
    padding: 10px;
}

.box {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    width: 755px;
    padding-right: 30px;
    /* box-shadow: inset 0 0 1em #f15e742f, 0 0 1em rgba(0, 0, 0, 0.233); */
    position: absolute;
    top: 17%;
    left: 21%;
    background-color: #00000065;
    border-radius: 15%;
    /* background: linear-gradient(-45deg, #ee7752, #e73c7e, #23a6d5, #23d5ab);
	background-size: 400% 400%;
	 animation: gradient 15s ease infinite; */
    /* background-image: url("https://png.pngtree.com/thumb_back/fw800/background/20190728/pngtree-purple-abstract-4k-free-wallpaper-background-download-png-and-psd-files-image_285854.jpg"); */
    /* padding: 5%; */
}
.banner  {
    display: flex;
    width: 100%;
    justify-content:space-between;
    margin-bottom: 10%;
    border-radius: 10%;
}
  
.banner img {
    width: 30%;
}

.title-container {
    display:flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.title {
    font-family: "Aquatico";
    font-size: 25pt;
}

.subtitle {
    font-family: 'Averia Sans Libre', cursive;
    font-size: 15pt;
}

.input {
    width: 250px;
    height: 40px;
    margin-top: 5px;
    border: none;
    font-family: "Work Sans", sans-serif;
    font-size: 12pt;
    padding: 2%;
    border-radius: 5px;
    background-color: #f3f3f3;
    margin:2% 0;
    cursor: pointer;
}

.input:hover {
    border: solid 1px #000;
}

.form {
    text-align: center;
    margin-top: 5%;
}

.buttons {
    width: 30%;
    height: 40px;
    margin-top: 3%;
    cursor: pointer;
    border: none;
    border-radius: 15px;
    font-family: 'Averia Sans Libre', cursive;
    font-size: 12pt;
    font-weight: bolder;
}

.signin {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.signin p {
    font-family: 'Averia Sans Libre', cursive;
    font-size: 15pt;
    font-weight: bolder;
    margin: 5% 0 5% 0;
}

.signin-button {
    border: solid 1px #000;
    transition: 0.5s ease-in-out;
}

.signin-button:hover{
    background-color: #FFCA19;
}

.signup-button {
  background-color: rgba(0, 0, 255, 0);
  border: none;
  cursor: pointer;
  font-size: larger;
  margin-top: -1%;
  color: #f3f3f3;
  width: 103%;
  font-size: 15pt;
  font-weight: lighter;
}

span{
  color: rgb(54, 54, 54);
  transition: 0.5s ease-in-out;
}

span:hover {
    color: #f3f3f3;
}

.google-button {
    width: 90%;
    display: flex;
    justify-content: space-around;
    align-items: center;
    margin: 0 5% 5% 5%;
    background-color: rgba(0, 0, 0, 0.65);
    border: solid 1px #000;
    transition: 0.5s ease-in-out;
    font-family: "Work Sans", sans-serif;
    font-weight: 700;
    color: #FFFF;
}

.google-button:hover {
    background-color: rgba(255, 255, 255, 0.863);
    color: #000;
}

.google-icon-btn {
    width: 10%;
}

.firebaseui-auth-container {
    margin: 5% 0 15% 0;
}