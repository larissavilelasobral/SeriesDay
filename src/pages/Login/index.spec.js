import Login from './index.js';
import * as services from '../../services/index.js';

services.googleLogin = jest.fn(() => Promise.resolve(true));
services.registerUser = jest.fn(() => Promise.resolve(true));

describe('Login', () => {
  it('should be a function', () => {
    expect(typeof Login).toBe('function');
  });

  it('should load the page', () => {
    expect(Login()).toMatchSnapshot();
  });

  // it('When the user clicks the button, he should call sign with email and password', () => {
  //   Login().querySelector('#button-login').dispatchEvent(new Event('click'));
  //   expect(services.signIn).toHaveBeenCalled();
  // });

  // it('When the user clicks the button he should call sign with google account', () => {
  //   Login().querySelector('#button-google').dispatchEvent(new Event('click'));
  //   expect(services.loginWithGoogle).toHaveBeenCalled();
  // });

  // it('When the user clicks the button, they should redirect to the registration page', () => {
  //   Login().querySelector('#button-create-account').dispatchEvent(new Event('click'));
  //   expect(utils.onNavigate).toHaveBeenCalledWith('/register');
  // });
});

//   describe('Login', () => {
//     it('should be a function', () => {
//       expect(typeof Login).toBe('function');
//     });

//     it('should load the page', () => {
//       expect(Login()).toMatchSnapshot();
//     });

//     it('should have a button with click event to sign in with email and password', () => {
//       Login().querySelector('#btn-login').dispatchEvent(new Event('click'));
//       expect(services.emailAndPasswordLogin).toHaveBeenCalled();
//       expect(services.emailAndPasswordLogin()).toBe(true);
//     });

//     it('should have a button with click event to sign in with google account', () => {
//       Login().querySelector('#btn-google').dispatchEvent(new Event('click'));
//       expect(services.googleLogin).toHaveBeenCalled();
//       expect(services.googleLogin()).toBe(true);
//     });
//     it('should have a button with click event to redirect user to sign up page', () => {
//       Login().querySelector('#sign-up-login').dispatchEvent(new Event('click'));
//       expect(utils.onNavigate).toBeCalledWith('/signup');
//     });
//   });

// importamos a função que vamos testar
// import { myFunction } from '../src/lib/index';

// describe('myFunction', () => {
//   it('should be a function', () => {
//     expect(typeof myFunction).toBe('function');
//   });
// });

/*
import { registerAccount } from '../src/lib/auth.js';

describe('Register', () => {
  it('should be a function', () => {
    expect(typeof registerAccount).toBe('function');
  });
  it('shold call firebase', () => {
    registerAccount();
    expect(firebase.auth().createUserWithEmailAndPassword()).toBeCalled();
  });
});
*/
