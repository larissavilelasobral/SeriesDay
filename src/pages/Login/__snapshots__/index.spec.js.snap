// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should load the page 1`] = `
<div>
  
    
  <input
    autocomplete="on"
    class="input"
    id="email"
    placeholder="E-mail"
    required=""
    type="email"
  />
  
    
  <input
    autocomplete="on"
    class="input"
    id="password"
    placeholder="Senha"
    required=""
    type="password"
  />
  
    
  <button
    class="signin-button buttons"
    id="signin-button"
  >
    Entrar
  </button>
  

    
  <button
    class="google-button buttons"
    id="google-button"
  >
    
        
    <img
      alt="Ícone do Google"
      class="google-icon-btn"
      src="../../assets/google-icon.svg"
    />
    
            Entrar com Conta Google
    
  </button>
  
    
</div>
`;
